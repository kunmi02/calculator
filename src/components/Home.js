import React from 'react';
import Navbar from './Navbar';

const Home = () => (
  <>
    <Navbar />
    <h2>Welcome to our Math Magician page!</h2>
    <p>
      Create some magic using our Mathematics calculatore
    </p>
    <p>
      Enjoy and catch fun
    </p>
  </>
);

export default Home;
