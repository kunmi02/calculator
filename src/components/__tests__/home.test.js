import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Home from '../Home';

const history = createBrowserHistory();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <Home />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains Title', () => {
  const { container, getByText } = render(
    <Router history={history}>
      <Home />
    </Router>,
  );
  expect(getByText('Welcome to our Math Magician page!')).toBeInTheDocument();
});

it('contains NavBar', () => {
  const { getByText } = render(
    <Router history={history}>
      <Home />
    </Router>,
  );
  expect(getByText('Math Whizz!!')).toBeInTheDocument();
});