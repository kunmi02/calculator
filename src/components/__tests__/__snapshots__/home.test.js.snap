// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="top"
  >
    <h1>
      Math Whizz!!
    </h1>
    <div
      className="links"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>,
  <h2>
    Welcome to our Math Magician page!
  </h2>,
  <p>
    Create some magic using our Mathematics calculatore
  </p>,
  <p>
    Enjoy and catch fun
  </p>,
]
`;
